conanfile.py

```diff
@@ -95,10 +97,18 @@ def set_version(self):
     def checkMoldIsSupported(self):
         """ Mold is only tested on Linux with gcc and clang. In future support for icc may be added. """
         return self.settings.os == "Linux" and (self.settings.compiler == "clang" or self.settings.compiler == "gcc")
+    
+    def checkCCacheIsSupported(self):
+        """ CCache is fully supported on Linux and macOS with gcc and clang. """
+        return (self.settings.os == "Macos" or self.settings.os == "Linux") and \
```

Comment:
I think this may not be needed eventually as we should be able to get this working on Windows and with MSVC and Intel compilers too: https://github.com/ccache/ccache/wiki/MS-Visual-Studio

But let's leave it for now and focus on these to begin with.

Reply:
I explain why I made this choice; because in the ccache [documentation](https://ccache.dev/platform-compiler-language-support.html) I saw that MSVC has not A support level:
![image](https://github.com/Twon/Morpheus/assets/113254017/3f6714f4-842a-44f5-be1b-64938135609a)
Anyway, as you said let's leave it as it is for now. I am keeping this conversation open, in this way we are not going to forget the point.

Reply:
Ah right, I think I have seen this before.  Well, let's try it out and see if it's useful on Windows.  Hopefully we can tie in some capture of build metrics at some point to shape our decisions.